Sources/GRPCCore/Stream/RPCWriterProtocol.swift

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// A sink for values which are produced over time.
+public protocol RPCWriterProtocol<Element>: Sendable {
+  /// The type of value written.
+  associatedtype Element: Sendable
```

Comment:
Having both `Sendable` constraints here might be a bit limiting and something that we should conditionalise. I am mostly thinking about unit tests here.